golemcli/src/commands/status.rs

```diff
@@ -0,0 +1,603 @@
+                                    |subtasks: Vec<SubtaskInfo>| {
+                                        subtasks
+                                            .iter()
+                                            .filter(|subtask| {
+                                                mem::discriminant(&subtask.status)
+                                                    == mem::discriminant(&SubtaskStatus::Finished)
+                                            })
+                                            .count()
+                                    },
+                                )
+                        }),
+                )
+            })
+            .map(
+                |(total_subtasks, finished_subtasks)| match total_subtasks > 0 {
+                    true => Some(format!("{}/{}", finished_subtasks, total_subtasks)),
+                    false => None,
+                },
+            )
+            .from_err()
+    }
+}
+
+impl FormattedObject for FormattedGeneralStatus {
+    fn to_json(&self) -> Result<Value, Error> {
+        Ok(serde_json::to_value(&self)?)
+    }
+
+    fn print(&self) -> Result<(), Error> {
+        let mut section_builder = SectionBuilder::new( "  ");
+
+        section_builder
+            .new_section("General")
+            .entry(
+                "Process State",
+                &self.running_status.process_state.to_string(),
+            )
+            .new_subsection("Components Status")
+            .entry(
+                "docker",
+                self.running_status
+                    .component_statuses
+                    .docker_status
+                    .as_ref()
+                    .unwrap_or(&String::from("unknown")),
+            )
+            .entry(
+                "hyperdrive",
+                self.running_status
+                    .component_statuses
+                    .hyperdrive
+                    .as_ref()
+                    .unwrap_or(&String::from("unknown")),
+            )
+            .entry(
+                "client",
+                self.running_status
+                    .component_statuses
+                    .client
+                    .as_ref()
+                    .unwrap_or(&String::from("unknown")),
+            );
+
+        if self.running_status.component_statuses.hypervisor.is_some() {
+            section_builder.entry(
+                "hypervisor",
+                self.running_status
+                    .component_statuses
+                    .hypervisor
+                    .as_ref()
+                    .unwrap(),
+            );
+        }
+        let connection_status = if self.net_status.is_connected {
+            String::from("ONLINE")
+        } else {
+            String::from("OFFLINE")
+        };
+
+        section_builder
+            .end_section()
+            .entry("Golem version", &self.running_status.golem_version)
+            .entry("Node name", &self.running_status.node_name)
```

Comment:
I'd try to format output similarily to `settings show`

```
┌───────────────────────────────────────────────────────────────────────┬─────────────┬─────────────────────────┐
│  description [name]                                                   │  value      │  type                   │
├───────────────────────────────────────────────────────────────────────┼─────────────┼─────────────────────────┤
│                                                                       │             │                         │
│  General                                                              │             │                         │
│                                                                       │             │                         │
│  Node name [node_name]                                                │  2rec iMac  │  str                    │
```

it looks much more readable and clear than

```
General
  Process State: running
  Components Status:
    docker: Docker is available
    hyperdrive: Connected to hyperg
    client: Connecting to network
    hypervisor: Docker VM is available
  Golem version: 0.19.2+dev603.g64869c2
  Node name: 2rec iMac
```

Reply:
I make it more readable without table. A table takes too much space.

Reply:
thx. I'd be great if you can show how it looks in tabular view, to compare both layouts